common/src/rc.rs

```diff
@@ -0,0 +1,14 @@
+#[cfg(not(feature = "threading"))]
+use std::rc::{Rc, Weak};
+#[cfg(feature = "threading")]
+use std::sync::{Arc, Weak};
+
+// type aliases instead of newtypes because you can't do `fn method(self: PyRc<Self>)` with a
+// newtype; requires the arbitrary_self_types unstable feature
+
+#[cfg(feature = "threading")]
+pub type PyRc<T> = Arc<T>;
```

Comment:
Do we still use `Py` prefix if when these code is regarded as non-python specific?

Reply:
I did it moreso to recognize that it's part of rustpython; I think it would be weird to just call it `Rc` when it could actually be an `Arc` as well.